app/server/src/features/accounts/methods.ts

```diff
@@ -179,6 +204,62 @@ export async function updatePassword(prisma: PrismaClient, input: UpdatePassword
     return { updatedUser, token };
 }
 
+/**
+ * returns token used to reset password
+ */
+export async function getResetPasswordToken(prisma: PrismaClient, input: ForgotPassRequestForm) {
+    const { email } = input;
+    
+    // fetch user for password validation
+    const user = await prisma.user.findUnique({ where: { email } });
+    if (!user) throw new Error('Request failed: No user exists with this email. Please input a different email.');
+
+    // generate token
+    const token = await jwt.sign({ email });
+
+    // return token
+    return { token };
+}
+
+/**
+ * resets the user's password using email retrieved from token
+ */
+export async function resetPassword(prisma: PrismaClient, input: ResetPasswordForm) {
+    const {
+        token,
+        newPassword,
+        confirmNewPassword
+    } = input;
+
+    // fetch email from token validation
+    const promise = await jwt.verify(token);
+    const result = JSON.parse(JSON.stringify(promise))
+    if (!result.email) throw new Error('Invalid token.');
+    
+    // fetch user for password validation
+    const user = await prisma.user.findUnique({ where: { email: result.email } });
+    if (!user) throw new Error('Resetting password failed: No user exists with this email. Please input a different email.');
```

Comment:
This is insecure.  Take a look at https://cheatsheetseries.owasp.org/cheatsheets/Authentication_Cheat_Sheet.html#password-recovery and change messages appropriately.